lib/src/main/kotlin/at/bitfire/ical4android/validation/FixInvalidDayOffsetPreprocessor.kt

```diff
@@ -14,7 +14,8 @@ object FixInvalidDayOffsetPreprocessor : StreamPreprocessor() {
         // Examples:
         // TRIGGER:-P2DT
         // TRIGGER:-PT2D
-        "^(DURATION|TRIGGER):-?P((T-?\\d+D)|(-?\\d+DT))\$",
+        // REFRESH-INTERVAL;VALUE=DURATION:PT1D
+        "(?:^|(^(DURATION|REFRESH-INTERVAL|RELATED-TO|TRIGGER));VALUE=)(DURATION|TRIGGER):-?P((T-?\\d+D)|(-?\\d+DT))\$",
```

Comment:
Last question (I hope): I have tried without `?:` and it worked as well.

I think it doesn't really matter whether we use it, because the strings don't contain `PT` or `DT`.

But let's assume there would be a property `CAPTION` with potential `VALUE=DURATION` that should also be fixed – in this case it would rewrite the property name to `CATION`, right (because of `.replace("PT", "P")``?)

So it would probably make sense to mark _everything_ except the actual capturing group that should be rewritten as non-capturing?

Reply:
> Last question (I hope): I have tried without `?:` and it worked as well.

How did you change the regex? Can you give a whole example? 

This: 
`"^|(^(DURATION|REFRESH-INTERVAL|RELATED-TO|TRIGGER));VALUE=(DURATION|TRIGGER):-?P((T-?\\d+D)|(-?\\d+DT))\$"`

fails for all tests with "Text cannot be parsed to a Duration", which makes sense I guess, since the whole string is not a valid definition of a duration.

Reply:
Ok what I meant is, if we use this regex: `(?:^|(^(CAPTION|DURATION|REFRESH-INTERVAL|RELATED-TO|TRIGGER));VALUE=)(DURATION|TRIGGER):-?P((T-?\\d+D)|(-?\\d+DT))\$` – I have only added the made-up property CAPTION here, just for illustration of the hypothetical problem – we get:

result: `CAPION;VALUE=DURATION:P1D`  – note the `CAPION`

Which is unexpected because I just added CAPTION.

However we don't have to add CAPTION or any other property that contains `PT` or `DT`. 😄  So it seems fine.

---

The other thing is, I don't get the usage of the non-capturing group yet. This regex works here, too:

```
override fun regexpForProblem() = Regex(
        "(^|(^(DURATION|REFRESH-INTERVAL|RELATED-TO|TRIGGER));VALUE=)(DURATION|TRIGGER):-?P((T-?\\d+D)|(-?\\d+DT))\$",
        setOf(RegexOption.MULTILINE, RegexOption.IGNORE_CASE)
    )
```

Because we don't use the captured groups at all, but replace `DT` and `PT` in the full match (which is basically the core of the "hypothetical problem" mentioned above):

```
        for (match in found) {
            val matchStr = match.value   // we operate on the full match here, not on captured groups
            // for illustration, we can use this here (output is then shown in test logs):
            System.err.println("Replacing PT and DT in $matchStr (${match.groupValues.size} captured groups)")
            val fixed = matchStr
                .replace("PT", "P")
                .replace("DT", "D")
```

---

I also just noticed that `match` has a `match.range` and we can directly use it to call replace: `s.replace(match.range, fixed)`. If I understand it correctly, this should replace the actually found occurrence and not run a new search.